keel-intents/src/main/kotlin/com/netflix/spinnaker/keel/intents/processors/converters/ApplicationConverter.kt

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.keel.intents.processors.converters
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spinnaker.keel.front50.Front50Service
+import com.netflix.spinnaker.keel.front50.model.Application
+import com.netflix.spinnaker.keel.intents.*
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import kotlin.reflect.KClass
+import kotlin.reflect.full.primaryConstructor
+
+@Component
+class ApplicationConverter(
+  private val front50Service: Front50Service,
+  private val objectMapper: ObjectMapper
+) : SpecConverter<BaseApplicationSpec, Application> {
+
+  private val log = LoggerFactory.getLogger(javaClass)
+
+  override fun convertToState(spec: BaseApplicationSpec): Application {
+
+    val application = Application(
+      name = spec.name,
+      description = spec.description.orEmpty(),
+      email = spec.email,
+      platformHealthOnly = spec.platformHealthOnly,
+      platformHealthOnlyShowOverride = spec.platformHealthOnlyShowOverride,
+      owner = spec.owner
+    )
+    application.apply {
+      set("chaosMonkey", spec.chaosMonkey)
+      set("enableRestartRunningExecutions", spec.enableRestartRunningExecutions)
+      set("instanceLinks", spec.instanceLinks)
+      set("instancePort", spec.instancePort)
+      set("appGroup", spec.appGroup)
+      set("cloudProviders", spec.cloudProviders)
+      set("accounts", spec.accounts)
+      set("dataSources", spec.dataSources)
+      set("requiredGroupMembership", spec.requiredGroupMembership)
+      set("group", spec.group)
+      set("providerSettings", spec.providerSettings)
+      set("trafficGuards", spec.trafficGuards)
+      set("platformHealthOnlyShowOverride", spec.platformHealthOnlyShowOverride)
+      set("platformHealthOnly", spec.platformHealthOnly)
+      set("notifications", spec.notifications)
+    }
+
+    if (spec is NetflixApplicationSpec) {
```

Comment:
We should chat tomorrow about how to start supporting extension / cloud-provider specific specs - I had to do similar things in the security group stuff. Definitely going to become a sore spot for keel pretty quickly.

Reply:
Yup, I agree. My thought is to get a starting implementation in first, so we don't launch a million orchestrations, then refactor this when we have a better plan.

Reply:
💯 Definitely not something blocking this PR.